src/ExpressiveInstaller/Resources/config/container-pimple.php

```diff
@@ -16,37 +17,41 @@
     && is_array($config['dependencies']['services'])
 ) {
     foreach ($config['dependencies']['services'] as $name => $service) {
-        $container[$name] = function ($c) use ($service) {
+        $container[$name] = function (Container $c) use ($service) {
             return $service;
         };
     }
 }
 
 // Inject factories
 foreach ($config['dependencies']['factories'] as $name => $object) {
-    $container[$name] = function ($c) use ($object, $name) {
-        if ($c->has($object)) {
-            $factory = $c->get($object);
+    $container[$name] = function (Container $c) use ($object, $name) {
+        $psrContainer = new PsrContainer($c);
+
+        if ($psrContainer->has($object)) {
+            $factory = $psrContainer->get($object);
         } else {
             $factory = new $object();
-            $c[$object] = $c->protect($factory);
+            $container[$object] = $c->protect($factory);
         }
 
-        return $factory($c, $name);
+        return $factory($psrContainer, $name);
     };
 }
 
 // Inject invokables
 foreach ($config['dependencies']['invokables'] as $name => $object) {
-    $container[$name] = function ($c) use ($object) {
+    $container[$name] = function (Container $c) use ($object) {
         return new $object();
     };
 }
 
 // Inject aliases
 foreach ($config['dependencies']['aliases'] as $alias => $target) {
-    $container[$alias] = function ($c) use ($target) {
-        return $c->get($target);
+    $container[$alias] = function (Container $c) use ($target) {
+        $psrContainer = new PsrContainer($c);
```

Comment:
Here we can avoid creating `PsrContainer`, and then use below
```php
return $c[$target];
```
What do you think @xtreamwayz ?